mitosheet/src/components/taskpanes/ControlPanel/ControlPanelTaskpane.tsx

```diff
@@ -58,6 +58,8 @@ export const ControlPanelTaskpane = (props: ControlPanelTaskpaneProps): JSX.Elem
     const [operator, setOperator] = useState(columnFilters !== undefined ? columnFilters.operator : 'And');
     const [stepID, setStepID] = useState('');
 
+    const originalNumRows = useRef(props.sheetData?.numRows || 0);
+    const editedFilter = useRef(false)
```

Comment:
Why are these refs vs. state variables? I think state variables are much safer, as refs don't trigger rerenders automatically.

If you don't need to set the state (e.g. originalNumRows), then you can use the following syntax:
```
const [originalNumRows, ] = useState(props.sheetData?.numRows || 0)
```